app/helpers/redirect_helper.rb

```diff
@@ -63,14 +44,19 @@ def previous_base_path
         previously_drafted_item.base_path
     end
 
-    def create_redirect(from_path:, to_path:, routes:)
-      RedirectHelper.create_redirect(
-        publishing_app: payload[:publishing_app],
-        old_base_path: from_path,
-        new_base_path: to_path,
-        routes: routes,
-        callbacks: callbacks,
-      )
+    def redirects_for(routes, old_base_path, new_base_path)
+      routes.map do |route|
+        {
+          path: route[:path],
+          type: route[:type],
+          destination: route[:path].gsub(old_base_path, new_base_path)
+        }
+      end
+    end
+
+    def owner
+      previously_published_item.try(:document) ||
```

Comment:
Ah yes that makes sense. Yeah it's the nil object pattern but it feels like there's too much data on it.